### ISSUE TITLE
VCF spec GT and AD INFO field keys confusion

### ISSUE THREAD

**Issue by tedtoal** [samtools/hts-specs#402]
There seems to be confusion surrounding the use of the GT and AD keys in the INFO field.

The GATK Mutect2 tool would produce a VCF file containing the following GT:AD values for a sample containing a homozygous variant with alternate allele depth of 20 and no reads for other alleles:

`0/1:0,20`

This is contrary to my understanding of these keys based on the VCF spec.  I would have expected:

`1/1:0,20`

Apparently GATK developers would argue that if GT were 1/1, then AD in this case should be 20,20:

`1/1:20,20`

because they view the GT numbers as indicators of which allele each AD number should indicate the depth of, and not as actual genotype indicators.

Suppose a sample has depths 20,20,2,1 for alleles ACGT with reference allele A.  Which of these would one expect for the ref, alt, and GT:AD?

`ref alt GT:AD`
`A C,G,T 0/1/2/3:20,20,2,1`
`A C 0/1:20,20`

Does the AD key contain one value for each allele shown in the ref and alt columns?  Or, does it contain one value for each number shown in the GT key?

Does the GT key contain one number for each distinct read value found in the sample?  Or does it contain one number for each putative allele in the sample?  Is it acceptable for the number of GT values to vary, e.g. if one tumor genome locus seems to be diploid there would be two values and if another seems to be hexaploid there would be six values?

I would like to see these issues clarified in the VCF spec document.  I would further suggest that, in the sample VCF lines shown in section 1.1, the AD key be added to the examples.  Also, it would be a good idea to show examples of each key in section 1.6.2.

**Comment by yfarjoun** [comment]
1. The vcf spec deals with the format of the data not with its meaning, so a program could, if the author wishes to,  GT:AD values of 1/1:100,0 even if that makes no "sense". that would still be a vlid VCF and the spec will have no opinion about it. 

that said,

2. GATK's mutect2 will not emit 1/1 by design. The GT field isn't really designed for the somatic use-case, and there is little sense to try to infer additional information about the mutation (allelic fraction for example) from it. much better to look for other, more informative fields (like AF or AD)

**Comment by tedtoal** [comment]
Ok, I'm beginning to understand it better.  I still think it would be wise if the VCF spec said more about this.  Clearly, most people think of the GT field as being genotype.  After thinking about the somatic case in terms of tumors with their wild genomes, I realize you are right, the entire concept of "genotype" is tumors is questionable (what if you have 100 copies of a locus?).  And yet, people talk about such things as LOH in a tumor, so the concept isn't entirely discarded.  I think the VCF spec should at least mention such issues and recommend against interpreting the GT key as genotype in cases of tumors and the like.

**Comment by pd3** [comment]
The VCF specification is quite clear on what GT means - that the alleles listed in the GT field are present in the sample. Of course the interpretation of the field will differ depending on the context, whether you work with population variation, somatic mutations, ensembles of cells, single cells, etc.

The question as it stands in this issue asks why the GT field emit is not according to the expectation given the AD field. That's a Mutect-specific question and Mutect forum would be more appropriate for this discussion.

**Comment by tedtoal** [comment]
I have asked the question on the Mutect forum and received what I thought was an unsatisfactory answer.  

I agree with you, it seems to me the VCF spec says that the GT field is genotype, and there is only one definition of that word, as far as I know.  However, assigning a genotype is quite problematic in tumor tissue, as you could have copy number change (e.g. 100 alleles rather than 2), multiple clones of different genotypes and policy, and mixture of normal and tumor tissue, which all create confusion for assigning a genotype.  I think it is mainly for this reason that Mutect does not use the GT field as prescribed by the VCF spec.  The spec REQUIRES the GT field, otherwise perhaps Mutect would have omitted it.  For a required field, its use in all situations where a VCF will be generated should be clear in the spec, but isn't for tumors.  I'd say the spec warrants at least a mention of how the GT field might best be used for a tumor genome locus.

**Comment by pd3** [comment]
The GT field is not mandatory. The only constraint is that if it is present, it has to be first in FORMAT.

**Comment by tedtoal** [comment]
Right, GT is optional.

Yes, it should remain flexible, but use of the GT field in the way Mutect2 uses it is very confusing and I don't think it conforms to the standard.  Again, I think it would be a good idea if the spec mentioned something about use of GT in tumor data.  I would think a reasonable think to say is that genotype can be hard to determine when a genome has many amplifications and deletions plus a mixture of clones and intermixed tumor/normal, and in this situation it may be best to omit GT, but if not omitted, it should reflect the estimated genotype at the locus.

**Comment by yfarjoun** [comment]
I agree that using 0/1 in Mutect is probably an abuse of notation. I'm not sure what the right notation is, but probably the AF should be enough for most users. The main problem (as I see it) that htsjdk, as currently implemented, forces a genotype object to have "alleles" and the interpretation of said alleles is the GT field. So, perhaps the correct avenue is for htsjdk to implement a new kind of genotype object that isn't only thinking about germline genotypes, but also considers other genotypes as reasonable targets...

I realize that this is a bit of a runaround for you, and I'm sorry for that.

**Comment by tedtoal** [comment]
That sounds reasonable.  Nevertheless, a starting point might be VCF spec addition mentioning the basic problem and steering people away from using GT or at least using it as it was intended.  That might serve to push for development of another type of genotype object in htsjdk.

**Comment by d-cameron** [comment]
The tumour case is particularly problematic due to the presence of
subclonal populations. Dynamic ploidy can be hacked into somatic vcfs but
the specs currently can't handle non-integer genotype values.

Phasing is also problematic due to the complex phlogenies of the DNA
segments in the derivative chromosomes.
